js/source/geojson_source.js

```diff
@@ -138,14 +138,19 @@ GeoJSONSource.prototype = util.inherit(Evented, /** @lends GeoJSONSource.prototy
             options.data = JSON.stringify(data);
         }
 
+        // We use a 'worker key' that's tied to the current geojson data for
+        // this source, so that when we ask the worker for tiles, the request
+        // goes to the same worker that parsed/prepared the geojson.
+        var workerKey = options.url || options.data;
```

Comment:
It looks like we reference this `workerKey` against a map. Will this work of `options.data` is an object?

```js
var object = {};

object[{foo: true}] = 'foo';
object[{bar: true}] = 'bar';

object[{foo: true}] // => 'bar';
object['[Object object]'] // => 'bar'
```